tests/mocha/insertion_marker_test.js

```diff
@@ -0,0 +1,388 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+suite('InsertionMarkers', function() {
+  setup(function() {
+    this.workspace = new Blockly.Workspace();
+    Blockly.defineBlocksWithJsonArray([
+      {
+        "type": "stack_block",
+        "message0": "",
+        "previousStatement": null,
+        "nextStatement": null
+      },
+      {
+        "type": "row_block",
+        "message0": "%1",
+        "args0": [
+          {
+            "type": "input_value",
+            "name": "INPUT"
+          }
+        ],
+        "output": null
+      },
+      {
+        "type": "statement_block",
+        "message0": "%1",
+        "args0": [
+          {
+            "type": "input_statement",
+            "name": "STATEMENT"
+          }
+        ],
+        "previousStatement": null,
+        "nextStatement": null
+      }]);
+  });
+  teardown(function() {
+    this.workspace.dispose();
+    delete Blockly.Blocks['stack_block'];
+    delete Blockly.Blocks['row_block'];
+    delete Blockly.Blocks['statement_block'];
+  });
+  suite('Code Generation', function() {
```

Comment:
To make sure that you're serializing the non-insertion marker block, add the ID to the code generation. Use a simple ID for the non-insertion marker blocks. Then you can also search for "insertion" in the generated code to make sure that block wasn't serialized.

Reply:
Ok it should now be asserting that correct blocks were serialized! (not just that the correct /number/ of blocks were serialized).